common/transport/mqtt/src/test/java/org/thingsboard/server/transport/mqtt/MqttTransportHandlerTest.java

```diff
@@ -205,4 +213,20 @@ public void givenMessageQueue_whenProcessMqttMsgConcurrently_thenEnqueueRegularS
         messages.forEach((msg) -> verify(handler, times(1)).processRegularSessionMsg(ctx, msg));
     }
 
-}
\ No newline at end of file
+    @Test
+    public void givenMqttConnectMessageAndPublishProvisionImmediately_whenProcessMqttMsg(){
+        givenMqttConnectMessage_whenProcessMqttMsg_thenProcessConnect();
+
+        List<MqttPublishMessage> messages = new ArrayList<>();
+        // add provision subscribe message
+        messages.add(getMqttPublishMessageAndProvisionResponse());
+
+        messages.forEach((msg) -> handler.channelRead(ctx, msg));
+
+        assertThat(handler.address, is(IP_ADDR));
+        assertThat(handler.deviceSessionCtx.getChannel(), is(ctx));
+        assertThat(handler.deviceSessionCtx.isConnected(), is(false));
+        verify(handler, never()).doDisconnect();
+        verify(handler, times(1)).processConnect(any(), any());
```

Comment:
In order to verify that added functionality works in this case you need to add check like:

`verify(handler, times(1)).processProvisionSessionMsg(any(), any());`